File: socket_client/command.js
#!/usr/bin/env node
API_BASE = process.env.API_BASE;
TOKEN = process.env.TOKEN;
NAMESPACE = process.env.NAMESPACE;

const socket = require('socket.io-client')(`${API_BASE}${NAMESPACE}`, {extraHeaders: {
    authorization: TOKEN
  }});
const someDelay = 10;
socket.on('connect', function () {
    console.log('connected...');
    if (process.argv[2] && process.argv[3]) {
        console.log('sending ' + process.argv[2] + ': ' + process.argv[3]);
        socket.emit('command', {'automation':process.argv[2], 'command':process.argv[3]});
        setTimeout(() => {
            process.exit(0);  
        }, someDelay);
    } else {
        console.log('usage: ./client.js <automation> <command>');
        process.exit(1);
    }
});

File: socket_client/command.py
import socketio, sys, time, os

API_BASE = os.getenv('API_BASE', None)
NAMESPACE = os.getenv('NAMESPACE', None)
TOKEN = os.getenv('TOKEN', None)

args = sys.argv[1:]
name = args[0]
status = args[1]

# standard Python
sio = socketio.Client()

sio.connect('http://localhost:8080', namespaces=['/', '/automations'], headers={'authorization':TOKEN})

sio.emit('status', {'automation':str(name), 'status':str(status)}, namespace='/automations')
#sio.emit('detail', {'automation':str(name), 'detail':{'name':'vpn', 'value':str(status)}}, namespace='/automations')
#sio.emit('command', {'automation':str(name), 'command':str(status)}, namespace='/automations')
#sio.emit('log', {'automation':str(name), 'log':str(status)}, namespace='/automations')

time.sleep(5)
sio.disconnect()